Imath/ImathMatrix.h

```diff
@@ -229,61 +230,64 @@ template <class T> class Matrix22
 
-    IMATH_CONSTEXPR14 const Matrix22& invert (bool singExc = false);
+    IMATH_CONSTEXPR14 const Matrix22& invert (bool singExc);
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 const Matrix22& invert();
 
-    IMATH_CONSTEXPR14 Matrix22<T> inverse (bool singExc = false) const;
+    IMATH_CONSTEXPR14 Matrix22<T> inverse (bool singExc) const;
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Matrix22<T> inverse() const;
 
     //------------
     // Determinant
     //------------
 
-    constexpr T determinant() const;
+    IMATH_HOSTDEVICE constexpr T determinant() const;
 
     //-----------------------------------------
     // Set matrix to rotation by r (in radians)
     //-----------------------------------------
 
-    template <class S> const Matrix22& setRotation (S r);
+    template <class S> IMATH_HOSTDEVICE const Matrix22& setRotation (S r);
 
     //-----------------------------
     // Rotate the given matrix by r
     //-----------------------------
 
-    template <class S> IMATH_CONSTEXPR14 const Matrix22& rotate (S r);
+    template <class S> IMATH_HOSTDEVICE IMATH_CONSTEXPR14 const Matrix22& rotate (S r);
 
     //--------------------------------------------
     // Set matrix to scale by given uniform factor
     //--------------------------------------------
 
-    IMATH_CONSTEXPR14 const Matrix22& setScale (T s);
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 const Matrix22& setScale (T s);
 
     //------------------------------------
     // Set matrix to scale by given vector
     //------------------------------------
 
-    template <class S> IMATH_CONSTEXPR14 const Matrix22& setScale (const Vec2<S>& s);
+    template <class S>
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 const Matrix22& setScale (const Vec2<S>& s);
 
     //----------------------
     // Scale the matrix by s
     //----------------------
 
-    template <class S> IMATH_CONSTEXPR14 const Matrix22& scale (const Vec2<S>& s);
+    template <class S> IMATH_HOSTDEVICE IMATH_CONSTEXPR14 const Matrix22& scale (const Vec2<S>& s);
 
     //--------------------------------------------------------
     // Number of the row and column dimensions, since
     // Matrix22 is a square matrix.
     //--------------------------------------------------------
 
-    constexpr static unsigned int dimensions() { return 2; }
+    IMATH_HOSTDEVICE constexpr static unsigned int dimensions() { return 2; }
 
     //-------------------------------------------------
     // Limitations of type T (see also class limits<T>)
     //-------------------------------------------------
 
-    constexpr static T baseTypeMin() { return limits<T>::min(); }
-    constexpr static T baseTypeMax() { return limits<T>::max(); }
-    constexpr static T baseTypeSmallest() { return limits<T>::smallest(); }
-    constexpr static T baseTypeEpsilon() { return limits<T>::epsilon(); }
+    IMATH_HOSTDEVICE constexpr static T baseTypeMin() { return limits<T>::min(); }
```

Comment:
I can't remember why these constexpr haven't been replaced by the IMATH_CONSTEXPR14 macro?

Reply:
If it has not been replaced then it means that it is capable of being compiled as constexr under the c++11 standard. If they were to be replaced with IMATH_CONSTEXR14 then it would not be constexr for c++11, but can be for c++14.

These functions are only a return statement which allows for c++11 to compile them as constexpr.

Reply:
Oh! Thanks for the explanation!